Extend getting-started guide with new example for getting a post

ernest-rudnicki opened the pull request:
This PR adds new example for getting a single post item from the database in `Getting Started` guide.

This closes the issue #201.

Here are the screenshots of the UI:

First part:
<img width="1242" alt="Zrzut ekranu 2023-07-13 o 17 35 56" src="https://github.com/sgrif/diesel.rs-website/assets/37155981/b4a5213b-0823-43c7-b75d-0ba18f428c18">



Second part:
<img width="1249" alt="Zrzut ekranu 2023-07-13 o 17 36 07" src="https://github.com/sgrif/diesel.rs-website/assets/37155981/ed7fada6-c9b3-4430-a168-5e77e1e72d58">

weiznich commented on src/guides/getting-started.md:
```diff
@@ -573,6 +573,58 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
+    let connection = &mut establish_connection();
+
+    let post = connection
+        .transaction(|connection| {
```
There is no need to wrap that into a transaction as it is only a single query.

weiznich commented on src/guides/getting-started.md:
```diff
@@ -573,6 +573,58 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
+                .first(connection)
+                .optional()?; // This allows for returning an Option<Post>, otherwise it will throw an error
+
+            match post {
+                Some(post) => Ok(post),
+                None => Err(diesel::result::Error::NotFound),
+            }
+        })
+        .unwrap_or_else(|_: diesel::result::Error| panic!("Unable to find post {}", post_id));
+
+    println!("Post with id: {} has a title: {}", post.id, post.title);
```
I would rather like to do something like the following here:

```rust
match post {
     Some(post) => println!("Post with id: …"),
     None => println!("No post with id {} found")
}
```

ernest-rudnicki replied:
I had to add also a third case for error, but it looks a lot better this way

weiznich commented on src/guides/getting-started.md:
```diff
@@ -573,6 +573,58 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
+
+    let post = connection
+        .transaction(|connection| {
+            let post = posts.find(id).select(Post::as_select())
```
Maybe add an additional sentence to the text above to point to the API documentation of `QueryDsl` for additional methods? That should make it much easier to discover all the other existing methods.

weiznich commented on src/guides/getting-started.md:
````diff
@@ -573,6 +573,58 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
 ```
 
+Additionally let's implement a possibility of fetching a single post. We will display the post id with its title.
+Notice `optional()` call. This returns `Option<Post>` instead of throwing an error, which we can then use in our matching pattern.
````
Link the API documentation of this method.

weiznich commented on src/guides/getting-started.md:
```diff
@@ -573,6 +573,58 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
+
+::: code-block
+
+[src/bin/get_post.rs](https://github.com/diesel-rs/diesel/blob/2.1.x/examples/postgres/getting_started_step_3/src/bin/get_post.rs)
```
This should probably link the example from the master branch for now, as the example needs to be merged there.

weiznich reviewed (commented):
Thanks for working on that example. It already looks good, but I've left some comments with minor improvements to make it easier to follow the code + add some links for easily get more information. It would be great if you could in cooperate these changes.

ernest-rudnicki commented:
@weiznich Thanks for great comments!. I adjusted the examples and the article. I updated the screenshots also to reflect new changes. It looks a lot better now 😄

weiznich commented on src/guides/getting-started.md:
````diff
@@ -573,6 +573,59 @@ That's it! Let's try it out with `cargo run --bin publish_post 1`.
 ```
 
+Additionally let's implement a possibility of fetching a single post. We will display the post id with its title.
+Notice [`.optional()`] call. This returns `Option<Post>` instead of throwing an error, which we can then use in our matching pattern. For additional methods of `QueryDsl` refer to the [`documentation`]
+
+
+[`.optional()`]: https://docs.diesel.rs/2.1.x/diesel/result/trait.OptionalExtension.html#tymethod.optional
+[`documentation`]: https://docs.diesel.rs/2.1.x/diesel/query_dsl/trait.QueryDsl.html
````
I feel that's important to state here what's in that documentation:

```suggestion
Notice [`.optional()`] call. This returns `Option<Post>` instead of throwing an error, which we can then use in our matching pattern. For additional methods to modify the constructed select statements refer to the [`documentation`  of `QueryDsl`]


[`.optional()`]: https://docs.diesel.rs/2.1.x/diesel/result/trait.OptionalExtension.html#tymethod.optional
[`documentation` of `QueryDsl`]: https://docs.diesel.rs/2.1.x/diesel/query_dsl/trait.QueryDsl.html
```

ernest-rudnicki replied:
Thanks! Done, I added the commit 😄

<img width="1229" alt="Zrzut ekranu 2023-07-13 o 19 01 11" src="https://github.com/sgrif/diesel.rs-website/assets/37155981/d65a5a09-a6ae-4153-915a-a5684e1557fb">

weiznich reviewed (commented):
Thanks for the fast update. :+1: 

This looks much better, but I would like to slightly reword that single sentence about `QueryDsl`
